caas/kubernetes/provider/constraints.go

```diff
@@ -23,5 +27,80 @@ var unsupportedConstraints = []string{
 func (k *kubernetesClient) ConstraintsValidator(ctx context.ProviderCallContext) (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
-	return validator, nil
+	return &constraintsValidator{Validator: validator}, nil
+}
+
+type constraintsValidator struct {
+	constraints.Validator
+}
+
+func validateTopologyKeyTagFunc(key string, topologySpreadKey string, tagList []string) error {
+	var topologySpreadValue, otherValue *string
+	// Search for the topologySpreadValue for topologyKeyTag
+	for tag := range tagList {
+		if strings.HasPrefix(tagList[tag], topologySpreadKey) {
+			trimmedValue := strings.TrimPrefix(tagList[tag], topologySpreadKey+"=")
```

Comment:
Does this handle "a = b" with white space?
You are better off splitting on "=" and then using Trim. Or better, have the parse function below do the heavy lifting in one place and return a map of key values. Then the logic here all becomes much simpler.